web/turkey/survey/static/survey/js/auditors/recorded_time_disparity.js

```diff
@@ -0,0 +1,49 @@
+var AuditorRecordedTimeDisparity = {
+    start_date: new Date(),
+    hidden: null,
+    visibility_change: null,
+    on_focus_time: 0,
+    last_focus_time: this.start_date, // switch in focus
+    setup: function() {
+        if (typeof document.hidden !== "undefined") {
+            this.hidden = "hidden";
+            this.visibility_change = "visibilitychange";
+        } else if (typeof document.mozHidden !== "undefined") {
+            this.hidden = "mozHidden";
+            this.visibility_change = "mozvisibilitychange";
+        } else if (typeof document.msHidden !== "undefined") {
+            this.hidden = "msHidden";
+            this.visibility_change = "msvisibilitychange";
+        } else if (typeof document.webkitHidden !== "undefined") {
+            this.hidden = "webkitHidden";
+            this.visibility_change = "webkitvisibilitychange";
+        }
+    },
+    log_recorded_time_disparity: function (e) {
+        if(document[this.hidden]) {
+            var focus_change_time = (new Date()).getTime();
+            this.on_focus_time += focus_change_time - this.last_focus_time;  
+        } else {
+            this.last_focus_time = (new Date()).getTime(); 
+        }
+    },
+    submit_callable: function () {
+        return {
+            'milliseconds': this.start_date.getTime() - this.on_focus_time
```

Comment:
Not sure I get what recorded_time_disparity is supposed to be getting. This calculation just takes the millisecond timestamp that `AuditorRecordedTimeDisparity` was evaluated and subtracts the value of `this.on_focus_time`. Is it supposed to record total time minus time in focus? That would be 

`((new Date()).getTime() - this.start_date.getTime()) - this.on_focus_time`